Octokit/Models/Response/OauthToken.cs

```diff
@@ -31,6 +32,24 @@ public OauthToken(string tokenType, string accessToken, IReadOnlyList<string> sc
         /// </summary>
         public IReadOnlyList<string> Scope { get; protected set; }
 
+        /// <summary>
```

Comment:
I think we should either overload the constructor or simply extend the existing one and privatize all of the setters on the properties - unless there is a reason we need to mutate this outside of construction - having this builder-like pattern would protect us from unintentional mutation while allowing us to be able to test/mock this model.

Alternatively, we could use an interface and abstract the error info into a more generalized response interface but I can also see why you'd want to encapsulate the error info in the OauthToken response object and just leave it as is.

Thoughts?

```
 public OauthToken(string tokenType, string accessToken, IReadOnlyList<string> scope, string error, string errorDescription, string errorUri)
        {
            TokenType = tokenType;
            AccessToken = accessToken;
            Scope = scope;
            Error = error;
            ErrorDescription = errorDescription;
            ErrorUri = errorUri;
        }

        /// <summary>
        /// The type of OAuth token
        /// </summary>
        public string TokenType { get; private set; }

        /// <summary>
        /// The secret OAuth access token. Use this to authenticate Octokit.net's client.
        /// </summary>
        public string AccessToken { get; private set; }

        /// <summary>
        /// The list of scopes the token includes.
        /// </summary>
        public IReadOnlyList<string> Scope { get; private set; }

        /// <summary>
        /// Gets or sets the error code or the response.
        /// </summary>
        [Parameter(Key = "error")]
        public string Error { get; private set; }

        /// <summary>
        /// Gets or sets the error description.
        /// </summary>
        [Parameter(Key = "error_description")]
        public string ErrorDescription { get; private set; }

        /// <summary>
        /// Gets or sets the error uri.
        /// </summary>
        [Parameter(Key = "error_uri")]
        public string ErrorUri { get; private set; }
```

Reply:
No problem, works for me! I can't remember for sure (I made this PR a looong time ago :) ) but from looking at some of the other existing classes yes I assume I made them protected following the convention of the existing ones. Making the props private and extending the ctor seems to work though